src/components/CombinedFeaturePanel/editors/AutoEditor.tsx

```diff
@@ -71,114 +70,82 @@ export const AutoEditor = ({
-      t` Your changes will still be visible on Open Street Map.`,
-    ];
-    toast.warning(message);
-    // const newPath = router.asPath.replace(new RegExp(`/edit/${tagName}`), '')
-    // router.push(newPath)
-  }, []);
+  function postErrorMessage() {
+    toast.error(
+      t`Something went wrong. Please let us know if the error persists.`,
+    );
+  }
 
-  const handleError = React.useCallback((error: Error, message?: string) => {
-    const defaultMessage = [
-      t`Your contribution could not be saved completely.`,
-      t`Please try again later or let us know if the error persists. Error: ${error}`,
-    ].join(" ");
+  const onLanguageChange = React.useCallback(
+    (newPickerValue: string) => {
+      const { normalizedOSMTagKey: baseTag } =
+        normalizeAndExtractLanguageTagsIfPresent(tagName);
+      const updatedTagName = [baseTag, newPickerValue].join(":");
 
-    toast.error(message || defaultMessage);
-  }, []);
+      if (updatedTagName !== finalTagName) {
+        setFinalTagName(updatedTagName);
+      }
+    },
+    [tagName, finalTagName],
+  );
 
-  const submitNewValue = useSubmitNewValueCallback({
-    handleSuccess,
-    handleOSMSuccessDBError,
-    handleError,
+  const updateTagValueWithLogIn = useUpdateTagValueWithLogInCallback({
     accessToken,
     baseUrl: remoteOSMAPIBaseUrl,
     osmType,
     osmId,
     tagName: finalTagName,
     newTagValue,
     currentOSMObjectOnServer: currentOSMObjectOnServer.data,
+    /* it would be more elegant to handle success or failure cases outside this hook, but in case the request to inhouse db
+    gets stuck, the entire function does not return and the user will not get any feedback for a while */
+    postSuccessMessage: postSuccessMessage,
+    postErrorMessage: postErrorMessage,
   });
 
-  const handleSubmitButtonClick = async () => {
+  const onSubmit = async () => {
     if (accessToken) {
-      await submitNewValue();
-      return;
-    }
-    if (!newTagValue || !osmId) {
-      handleError(
-        new Error("Missing Information"),
-        t`Some information was missing while saving to OpenStreetMap. Please let us know if the error persists.`,
-      );
+      try {
+        await updateTagValueWithLogIn();
+      } catch (error) {
+        log.error(error);
+      }
       return;
     }
+
     try {
-      await makeChangeRequestToInhouseApi({
+      await updateTagValueNoLogIn({
         baseUrl: inhouseOSMAPIBaseURL,
-        osmId,
+        osmType: osmType,
+        osmId: osmId,
         tagName: finalTagName,
-        newTagValue,
+        newTagValue: newTagValue,
+        /* it would be more elegant to handle success or failure cases outside this function, but in case the request to inhouse db
+        gets stuck, the entire function does not return and the user will not get any feedback for a while */
+        postSuccessMessage: postSuccessMessage,
+        postErrorMessage: postErrorMessage,
```

Comment:
suggestion (non-blocking): You can move the logic to check if newTagValue, osmType and osmId are properly set out of the hook and check them here to have the error handling in this file.